pcap/pcapio/reader.go

```diff
@@ -43,7 +51,9 @@ func NewReader(r io.Reader) (Reader, error) {
 	track.Reset()
 	_, err2 := NewNgReader(track)
 	if err2 == nil {
-		return NewNgReader(recorder)
+		r, _ := NewNgReader(recorder)
+		r.SetWarningChan(warningCh)
+		return r, nil
```

Comment:
Nit: Even though err should always be nil, I don't think there's a compelling reason not to capture and return it.
```suggestion
		r, err := NewNgReader(recorder)
		r.SetWarningChan(warningCh)
		return r, err
```